server/sandbox_run_linux.go

```diff
@@ -185,13 +185,12 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	}
 
 	// Add special container name label for the infra container
-	labelsJSON := []byte{}
 	if labels != nil {
 		labels[types.KubernetesContainerNameLabel] = leaky.PodInfraContainerName
-		labelsJSON, err = json.Marshal(labels)
-		if err != nil {
-			return nil, err
-		}
+	}
+	labelsJSON, err := json.Marshal(labels)
```

Comment:
I assume that marshaling `nil` converts to `{}`, right?

Reply:
Actually `nil` converts to `null`

Reply:
Ah okay, but wouldn't this cause to fail that line:
https://github.com/cri-o/cri-o/blob/92863e3b30774b6329ffa0790073d29b9ee9dc7b/internal/lib/container_server.go#L157

Reply:
No. It doesn't fail. `m.Annotations[annotations.Labels]` in `"null"` in that case. And that is valid JSON.

Reply:
Awesome, thanks for the clarification.